### PR TITLE
containers that only use env secrets also need the secret puller

### PR DESCRIPTION
**Pull request by grosser**

### CODE REVIEW

**Comment by grosser** [comment]
@ragurney

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 1]
```diff
@@ -392,17 +392,18 @@ def static_env
...
       expand_secret_annotations
     end
 
     def secret_env_as_annotations?
       ENV["SECRET_ENV_AS_ANNOTATIONS"]
     end
 
-    # storing secrets as env vars makes them visible in the deploy docs and when inspecting deployments, avoid it
+    # storing secrets as env vars makes them visible in the deploy docs and when inspecting deployments, to avoid it
     # we replace all secrtes from the env here and they are expanded by expand_secret_annotations later
```
Comment (adammw) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]: fix typo

File: plugins/kubernetes/test/models/kubernetes/template_filler_test.rb [hunk 2]
```diff
@@ -668,6 +672,13 @@ def add_init_contnainer_new_syntax(container)
             env.select { |e| ["BAR", "BAZ"].include?(e[:name]) }.must_equal [{name: "BAZ", value: "nope-secret://foo"}]
           end
         end
+
+        it "works when no other seret annotation was set" do
```
Comment (adammw) [inline plugins/kubernetes/test/models/kubernetes/template_filler_test.rb]: typo

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb [hunk 3]
```diff
@@ -392,17 +392,18 @@ def static_env
     end
 
     def set_secrets
-      return unless needs_secret_puller?
-      set_secret_puller
+      return unless SECRET_PULLER_IMAGE
       convert_secret_env_to_annotations if secret_env_as_annotations?
```
Comment (adammw) [inline plugins/kubernetes/app/models/kubernetes/template_filler.rb]: is this part tested?

Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/template_filler.rb]: yep

File: plugins/kubernetes/app/models/kubernetes/template_filler.rb (hunk 1 above)
Comment (grosser) [reply plugins/kubernetes/app/models/kubernetes/template_filler.rb]: ... was looking for that typo on the green line far too long :/

**Pull request merged**